api_initiative.py

```diff
@@ -57,21 +84,54 @@ def remove_entity_from_combat(self, args, username):
 
 def start_combat(self, args, username):
     campaign_id = args[KEY_CAMPAIGN_ID]
-    campaign = self.server.db.get_campaign(campaign_id)
-    role = campaign["members"][username]
-
-    if role == ROLE_GM and self.server.db.start_combat(campaign_id):
+    if has_gm_permissions(self, username, campaign_id) and self.server.db.start_combat(campaign_id):
         return self.respond({"success": True})
 
     return self.respond({"success": False, "info": MSG_NO_PERMISSION})
 
 
 def end_combat(self, args, username):
     campaign_id = args[KEY_CAMPAIGN_ID]
+    if has_gm_permissions(self, username, campaign_id) and self.server.db.end_combat(campaign_id):
+        return self.respond({"success": True})
+
+    return self.respond({"success": False, "info": MSG_NO_PERMISSION})
+
+
+def update_initiative_style(self, args, username):
+    campaign_id = args[KEY_CAMPAIGN_ID]
+    style = args[KEY_STYLE]
+    if style not in INIT_STYLES:
+        return self.respond({"success": False, "info": MSG_NO_INIT_STYLE})
+    if has_gm_permissions(self, username, campaign_id):
+        self.server.db.update_initiative_style(campaign_id, style)
+        return self.respond({"success": True})
+
+    return self.respond({"success": False, "info": MSG_NO_PERMISSION})
+
+
+# This endpoint basically just serves as an easy way to end an entity's turn.
+# Generally, entity's turns should be ended automatically after using all of their actions.

```

Comment:
I'd like to have explicit end turn calls no matter what to avoid complications with edge cases like "I want to use 3 Actions and then use my Reaction as something on my turn" or "I want to use my 3 Actions and then use a Hero point to get another Action back and keep my turn"

Reply:
Oh good point!
btw - this code is probably about down - there are 2 more endpoints I need to test / add to example.py and then maybe I'll fix up some documentation and this should be good to go.